src/app/components/form/input/FormInputEmailUsername.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+  <FormInput
+    v-if="formInput"
+    :is-optional="isOptional"
+    :id-label="`input-${id}`"
+    :title="title || t('usernameOrEmail')"
+    :type="inputType"
+    :value="formInput"
+    @input="handleInput"
+  >
+    <template #stateError>
+      <FormInputStateError
+        :form-input="formInput"
+        validation-property="lengthMax"
+      >
+        {{ t('globalValidationLength') }}
+      </FormInputStateError>
+      <FormInputStateError
+        :form-input="formInput"
+        validation-property="required"
+      >
+        {{ t('globalValidationRequired') }}
+      </FormInputStateError>
+      <FormInputStateError :form-input="formInput" validation-property="format">
+        {{ t('globalValidationFormat') }}
+      </FormInputStateError>
+    </template>
+    <template #stateWarning>
+      <FormInputStateWarning
+        v-if="
+          formInput.$dirty &&
+          formInput.$model &&
+          ((formInput.$model.includes('@') &&
+            !isValidEmail(formInput.$model)) ||
+            (!formInput.$model.includes('@') &&
+              !isValidUsername(formInput.$model)))
+        "
+      >
+        {{ t('globalValidationCheck') }}
+      </FormInputStateWarning>
+    </template>
+  </FormInput>
+</template>
+
+<script setup lang="ts">
+import type { BaseValidation } from '@vuelidate/core'
+import { email } from '@vuelidate/validators'
+import { VALIDATION_FORMAT_SLUG } from '~/utils/validation'
+import { computed } from 'vue'
```

Comment:
```suggestion

import { VALIDATION_FORMAT_SLUG } from '~/utils/validation'
```

No need to import from `vue`, is auto-imported.